application/src/qmlengineinterceptor.cpp

```diff
@@ -1,80 +1,141 @@
+                }
+            }
+
+            if ( !importSegment.isEmpty() ){
+                QStringList importSegmentParts = importSegment.split('/');
+                std::vector<std::string> partsConverted;
+                for ( auto it = importSegmentParts.begin(); it != importSegmentParts.end(); ++it ){
+                    int dotIndex = it->indexOf('.');
+                    if ( dotIndex != -1 ){
+                        partsConverted.push_back(it->mid(0, dotIndex).toStdString());
+                    } else {
+                        partsConverted.push_back(it->toStdString());
+                    }
+                }
+
+                try{
+                    Plugin::Ptr plugin = m_packageGraph->loadPlugin(partsConverted);
+                    if ( plugin != nullptr ){
+                        return QUrl::fromLocalFile(QString::fromStdString(plugin->path() + "/qmldir"));
+                    }
+                } catch ( lv::Exception& ){}
+            }
+        } catch ( lv::Exception& e ){
+            m_engine->throwError(&e, nullptr);
+        }
+    }
+    else if (dataType == QQmlAbstractUrlInterceptor::QmlFile && path.isLocalFile()){
+        if (m_project->isOpened(path.toLocalFile())){
+            QString str = path.toString();
+            if (str.length() >= 4 && str.left(4) == "file")
+            {
+                return QUrl("memory" + str.right(str.length()-4));
+            }
+            return path;
+        }
+
+    }
+    return path;
+}
+
+QNetworkAccessManager *QmlEngineInterceptor::Factory::create(QObject *parent){
+    return new QmlEngineInterceptor(parent);
+}
+
+QNetworkReply *QmlEngineInterceptor::createRequest(
+        QNetworkAccessManager::Operation op,
+        const QNetworkRequest &request,
+        QIODevice *outgoingData)
+{
+    if (request.url().scheme() == "memory")
+    {
+        MemoryNetworkReply* mnr = new MemoryNetworkReply();
+        QString content = m_project->openFile(request.url().toLocalFile(), ProjectDocument::Monitor)->content();
+        mnr->setContent(content);
+        return mnr;
+    }
+    return QNetworkAccessManager::createRequest(op, request, outgoingData);
+}
+
+
+void QmlEngineInterceptor::interceptEngine(ViewEngine *engine, PackageGraph* packageGraph, Project* project){
+    m_project = project;
+    engine->engine()->setUrlInterceptor(new QmlEngineInterceptor::UrlInterceptor(engine, packageGraph, project));
+}
+
+void MemoryNetworkReply::setContent(const QString &content)
+{
+    setAttribute(QNetworkRequest::HttpStatusCodeAttribute, 200);
+    setAttribute(QNetworkRequest::HttpReasonPhraseAttribute, "OK");
+    setHeader(QNetworkRequest::ContentTypeHeader, "text/html");
+    m_content = content.toUtf8();
+    m_offset = 0;
+    open(ReadOnly | Unbuffered);
+    setHeader(QNetworkRequest::ContentLengthHeader, QVariant(content.size()));
+    QTimer::singleShot(0, this, SIGNAL(readyRead()));
+    QTimer::singleShot(0, this, SIGNAL(finished()));
```

Comment:
Am not sure these 2 signals are needed. TODO: check wether this works without them.

Use `QNetworkReply::setFinished(true)` instead of `finished()` signal.
`bytesAvailable()` should return wether this reply has any bytes for reading, it's not clear wether the signal is required in this case.